zmon-controller-app/src/main/java/org/zalando/zmon/controller/MultiKairosDBController.java

```diff
@@ -89,15 +115,59 @@ public MultiKairosDBController(KairosDBProperties kairosDBProperties, MetricRegi
             q.put("cache_time", 60);
             if (q.has("start_absolute")) {
                 long start = q.get("start_absolute").asLong();
+                long finish = q.get("end_absolute").asLong();
+                queryDistance = (System.currentTimeMillis() - start)/86400000;
+                querySpan = (finish-start)/86400000;
                 start = start - (start % 60000);
                 q.put("start_absolute", start);
             }
             else if (q.has("start_relative")) {
+
+                ObjectNode r = (ObjectNode)q.get("start_relative");
+
                 if(queryWindow != 0) {
-                    ObjectNode r = (ObjectNode)q.get("start_relative");
                     r.put("value", queryWindow);
                     r.put("unit", "minutes");
                 }
+
+                switch(r.get("unit").asText()){
```

Comment:
AFAIU unit is always present and defaults to dashboard/widget setting. So it could be hours, minutes, etc. 

unit is set to minutes in the code if queryWindow != 0. It defaults to 0 and don't know yet whether it is set to a non zero value anywhere.